File: sem4/MPP/proiectC#/AppClientServer/Model/Child.cs
using System;

namespace model
{
    [Serializable]
    public class Child: Entity<int>
    {
        public string name { get; set; }
        public int age { get; set; }

        public Child(string name, int age)
        {
            this.name = name;
            this.age = age;
        }

        public override bool Equals(object? obj)
        {
            if (obj is Child)
            {
                return (obj as Child).ID == ID;
            }
            return base.Equals(obj);
        }

        public override string ToString()
        {
            return ID + " " + name + " " + age;
        }
    
        public static bool operator ==(Child c1, Child c2)
        {
            return c1.Equals(c2);
        }
    
        public static bool operator !=(Child c1, Child c2)
        {
            return !c1.Equals(c2);
        }  
    }
}

File: sem4/MPP/proiect/appClientServer/CompetitionModel/src/main/java/org/example/model/Child.java
package org.example.model;

import java.io.Serializable;
import java.util.Objects;

public class Child extends Entity<Integer> implements Serializable {
    private String name;
    private int age;

    public Child(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public Child() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    @Override
    public String toString() {
        return "Child{" +
                "name='" + name + '\'' +
                ", age=" + age +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Child child = (Child) o;
        return age == child.age && name.equals(child.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, age);
    }
}
